coalib/processes/Processing.py

```diff
@@ -421,6 +429,31 @@ def instantiate_processes(section,
             bear_runner_args)
 
 
+def uncache_if_unloaded_bears(initial_count,
+                              loaded_count,
+                              files_dict,
+                              cache,
+                              log_printer):
+    """
+    Checks if any bear could not be instantiated because of a missing
+    dependency and uncaches current section's cached files
+    to permit a future execution of the failing bear.
+
+    :param initial_count: The number of bears that the section should execute.
+    :param loaded_count:  The number of bears that were loaded with success.
+    :param files_dict:    The files this section runs on. These are the files
+                          that will be removed from the cache only in case
+                          there was a problem instantiating all bears right.
+    :param cache:         An instance of ``misc.Caching.FileCache`` to use as
+                          a file cache buffer.
+    :param log_printer:   The log printer to warn to.
+    """
+    if cache and initial_count != loaded_count:
+        log_printer.debug('The cache has been reset because not all bears '
+                          'could be instantiated.')
+        cache.untrack_files(files_dict)
```

Comment:
Don't understand that :/ 

> There is no test written for that function

You have a test, haven't you?

> and i could not get 100% coverage when i let the code there

really? the current tests should cover the if-skipping (as the condition does not hold for normal coala operation), and your test tries to go into the if-condition and untracks the cache. It should have 100% I believe^^